File: Hello/step07sqlite/src/main/java/com/example/step07sqlite/DBHelper.java
package com.example.step07sqlite;

import android.content.Context;
import android.database.sqlite.SQLiteDatabase;
import android.database.sqlite.SQLiteOpenHelper;

import androidx.annotation.Nullable;

/*
    DB 생성 및 리셋을 도와주는 도우미 클래스 만들기
    - SQLiteOpenHelper 추상 클래스를 상속 받아서 만든다.
 */
public class DBHelper extends SQLiteOpenHelper {
    //생성자
    public DBHelper(@Nullable Context context, @Nullable String name,
                    @Nullable SQLiteDatabase.CursorFactory factory, int version) {
        super(context, name, factory, version);
    }
    //App 에서 DB 를 처음 사용할때 호출되는 메소드
    @Override
    public void onCreate(SQLiteDatabase db) {
        //사용할 테이블을 만들면 된다.
        String sql="CREATE TABLE todo " +
                "(num INTEGER PRIMARY KEY AUTOINCREMENT, " +
                "content TEXT, regdate TEXT)";
        //SQLiteDataBase 객체를 이용해서 실행한다.
        db.execSQL(sql);
    }

    @Override
    public void onUpgrade(SQLiteDatabase db, int i, int i1) {
        //업그래이드할 내용을 작성하면 된다.
        db.execSQL("DROP TABLE IF EXISTS todo");
        //다시 만들어 질수 있도록 onCreate() 메소드를 호출한다.
        onCreate(db);
    }
}

File: Hello/step07sqlite2/src/main/java/com/example/step07sqlite2/DBHelper.kt
package com.example.step07sqlite2

import android.content.Context
import android.database.sqlite.SQLiteDatabase
import android.database.sqlite.SQLiteOpenHelper
import java.util.prefs.PreferencesFactory

/*
    1. SQLiteOpenHeler 클래스는 추상 클래스 이기 때문에 추상메소드를 오버라이드 해야한다.
    2. DBHelper 클래스의 대표 생성자에서 부모 생성자에 전달할 값을 받아야한다.
    3. SQLiteOpenHelper 클래스의 생성자에 필요한 값을 전달해야 한다.
 */
class DBHelper(
    context : Context?,
    name : String?,
    factory : SQLiteDatabase.CursorFactory?,
    version : Int
) : SQLiteOpenHelper(context, name, factory, version){

    override fun onCreate(db: SQLiteDatabase?) {
        val sql="""
            CREATE TABLE todo
            (num INTEGER PRIMARY KEY AUTOINCREMENT,content TEXT, regdate TEXT)
        """.trimIndent()
        db?.execSQL(sql)
    }

    override fun onUpgrade(db: SQLiteDatabase?, p1: Int, p2: Int) {
        db?.execSQL("DROP TABLE IF EXISTS todo")
        onCreate(db)
    }

}